### PR TITLE
Adds docs for Display and Editor Templates

### LINKED ISSUE #25921: Offer to write introduction and usage for DisplayTemplates/EditorTemplates
The DisplayTemplates/EditorTemplates are a very useful system to abstract displaying values and form controls in views.

Unfortunately, they’re basically undocumented.

I would like to write an introduction to them and show how to use them, by also providing a minimal sample project (using the outlined conventions in the contribution guide of course) with a razor page that uses a DisplayTemplate and EditorTemplate for a small custom type.

The outline would be
* Use case
* How they work (folder/file name convention)
* How to use them with an example

If this is OK with you, please give me a green light, and I will make a pull request shortly.

Concerning the location of the page, perhaps within the _Web Apps_ /_Advanced_ section would be a good place.

Thanks!
Alex

### PR DESCRIPTION
**Pull request by Ducki**
Fixes #25921

This PR creates a new section "Display and Editor Templates" within _/mvc/views_.

It describes the use case and usage for the DisplayTemplates, which are used by the [DisplayFor](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.rendering.htmlhelperdisplayextensions.displayfor?view=aspnetcore-6.0) and [EditorFor](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.rendering.htmlhelpereditorextensions.editorfor?view=aspnetcore-6.0) HTML helper methods.

It also adds a reference to this section in the `toc.yml` file.

Additionally, it adds a minimal Razor Pages sample project in the appropriate directory, to showcase the usage in a working environment – and to be used by the code snippet references.

Please have a look at the YAML header of `display-templates.md`, I'm not sure if it contains all the necessary metadata, especially the missing 'ms.author' tag (I don't have one…).

Thank you very much! :)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=25952) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=25952) before we can accept your contribution.<br/><br/>:x: Ducki [sign now](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=25952)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/AspNetCore.Docs?pullRequest=25952) it.</sub>

File: aspnetcore/mvc/views/display-templates.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+title: Using DisplayTemplates and EditorTemplates
+author: Ducki
```
Comment (Rick-Anderson) [inline aspnetcore/mvc/views/display-templates.md:3]:
```suggestion
author: Ducki
ms.author: riande
```

**Comment by Rick-Anderson** [comment]
@Ducki  I pushed a commit to fix a warning so you'll have to pull before you push again.

**Comment by Rick-Anderson** [comment]
@Ducki  closing, I brought your commits to #25959

**Pull request closed without merging**